react/ButtonAction/index.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import styles from './styles.styl'
+import Icon from '../Icon'
+
+const usableByIcon = icon => {
+  const isSvgSymbol = icon && !!icon.id
+  const isIconIdentifier = typeof icon === 'string'
+  return isSvgSymbol || isIconIdentifier
+}
+
+const ButtonAction = props => {
+  const { type, disabled, children, label, leftIcon, rightIcon, compact, className, onClick, ...restProps } = props
+  return (
+    <button
+      disabled={disabled}
+      type="button"
+      role="button"
+      className={cx(
+        styles['c-actionbtn'], {
+          [styles[`c-actionbtn--${type}`]] : type,
+          [styles[`c-actionbtn--compact`]] : compact
+        },
+        className)}
+      onClick={onClick}
+      title={compact ? label : undefined}
+      {...restProps}
+    >
+      <span>
+        {usableByIcon(leftIcon) ? <Icon icon={leftIcon} /> : leftIcon}
+        {label &&
+          <span
+            data-action="label"
```

Comment:
I see that data-action is used to target this span in CSS, why not using a class ?

Reply:
Yeah I know, it seems odd, but I try very hard not using classes in stylus `$placeholder` so it isn't forced classes and it reduces the risk of name conflict. 
The thing is I cannot make parent > children relationship between placeholders but for this one I really needed it, so I add something as neutral as possible.

Reply:
> I try very hard not using classes in stylus $placeholder so it isn't forced classes